src/atc/delta/delta_handle.py

```diff
@@ -0,0 +1,95 @@
+from pyspark.sql import DataFrame
+
+from atc.atc_exceptions import AtcException
+from atc.config_master import TableConfigurator
+from atc.functions import init_dbutils
+from atc.spark import Spark
+
+
+class DeltaHandleException(AtcException):
+    pass
+
+
+class DeltaHandleInvalidName(DeltaHandleException):
+    pass
+
+
+class DeltaHandleInvalidFormat(DeltaHandleException):
+    pass
+
+
+class DeltaHandle:
+    def __init__(self, name: str, location: str = None, data_format: str = "delta"):
+        self._name = name
+        self._location = location
+        self._data_format = data_format
+
+        self._validate()
+
+    @classmethod
+    def from_tc(cls, id: str):
+        tc = TableConfigurator()
+        return cls(
+            name=tc.table_name(id),
+            location=tc.table_path(id),
+            data_format=tc.table_property(id, "format", "delta"),
+        )
+
+    def _validate(self):
+        # name is either `db`.`table` or just `table`
+        name_parts = self._name.split(".")
+        if len(name_parts) == 1:
+            self._db = None
+            self._table_name = name_parts[0]
+        elif len(name_parts) == 2:
+            self._db = name_parts[0]
+            self._table_name = name_parts[1]
+        else:
+            raise DeltaHandleInvalidName(f"Could not parse name {self._name}")
+
+        # only format delta is supported.
+        if self._data_format != "delta":
+            raise DeltaHandleInvalidFormat("Only format delta is supported.")
+
+    def read(self) -> DataFrame:
+        """Read table by path if location is given, otherwise from name."""
+        if self._location:
+            return Spark.get().read.format(self._data_format).load(self._location)
+        return Spark.get().table(self._name)
+
+    def write_or_append(self, df: DataFrame, mode: str, overwriteSchema=None) -> None:
+        writer = df.write.format(self._data_format).mode(mode)
+        if overwriteSchema is not None:
+            writer = writer.option("overwriteSchema", overwriteSchema)
+
+        if self._location:
+            return writer.save(self._location)
+
+        return writer.saveAsTable(self._name)
+
+    def overwrite(self, df: DataFrame, overwriteSchema=None) -> None:
+        return self.write_or_append(df, "overwrite", overwriteSchema=overwriteSchema)
+
+    def append(self, df: DataFrame, overwriteSchema=None) -> None:
+        return self.write_or_append(df, "append", overwriteSchema=overwriteSchema)
+
+    def truncate(self) -> None:
+        Spark.get().sql(f"TRUNCATE TABLE {self._name};")
+
+    def drop(self) -> None:
+        Spark.get().sql(f"DROP TABLE IF EXISTS {self._name};")
+
+    def drop_and_delete(self) -> None:
+        self.drop()
+        if self._location:
+            init_dbutils().fs.rm(self._location, True)
```

Comment:
This is might not a realistic case. But what if the table actually has a location when it was created - but has not self._location?

We have a function called "drop_table_cascade" that finds the location of the table, removes the files, and then drops the table. It is not necessary that it should be used in this context, but just to remind that the function exists. 
https://github.com/atc-net/atc-dataplatform/blob/main/src/atc/functions.py

Reply:
the other function is only marked as deprecated, not removed.